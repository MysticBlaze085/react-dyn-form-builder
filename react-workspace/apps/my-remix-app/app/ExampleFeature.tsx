/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unused-vars */

import React, { useCallback, useEffect, useState } from 'react';

import {FormBuilder} from 'shared-ui/node_modules/react-reactive-form';
import { FormGeneratorWrapper } from 'shared-ui';
import formInputConfig from './form.config';

const ExampleFeature = () => {
    const defaultGroup = FormBuilder.group({
        first_name: [],
        last_name: []
    })
    const [genForm, setGenForm] = useState<any>(defaultGroup)

    const setInputNesting = useCallback(() => {
        console.log('genForm', genForm.get('first_name'));
        genForm
            .get('first_name')
            .onBlurChanges.subscribe((value: any) => (value ? genForm.patchValue({ full_name: `${value} - ` }) : ''));
        genForm
            .get('last_name')
            .onBlurChanges.subscribe((value: any) => {
                const initial = genForm.get('first_name').value;
                genForm.patchValue({ full_name: `` });
                genForm.patchValue({ full_name: `${initial} - ${value}` })
            })
        genForm.disable();
    }, [genForm]);

    const handleSubmit = (e: any) => {
        e.preventDefault();
        alert(`You submitted \n ${JSON.stringify(genForm.value, null, 2)}`);
        console.log(genForm);
    };

    const handleEditMode = () => {
        genForm.enable({ onlySelf: true });
    };

    const handleOnCancelMode = () => {
        genForm.disable({ onlySelf: true });
        genForm.reset();
    };

    const setForm = (form: any) => {
        setGenForm({
            ...form,
            meta: {
                handleSubmit: handleSubmit,
                handleEditMode: handleEditMode,
                handleCancelMode: handleOnCancelMode,
            }
        });
    };

    const unSubscribe = useCallback(() => {
        genForm.valueChanges.unsubscribe();
    }, [genForm]);

    useEffect(() => {
        setInputNesting();
        unSubscribe();
    }, [setInputNesting, unSubscribe]);

    return (
        <div className="h-100">
            <div className="row">
                <FormGeneratorWrapper onMount={setForm} fieldConfig={formInputConfig} />
            </div>
        </div>
    );
};

export default ExampleFeature;
