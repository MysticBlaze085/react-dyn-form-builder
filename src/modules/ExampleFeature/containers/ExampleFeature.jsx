import React, { useCallback, useEffect } from 'react';
import { FormGenerator } from 'react-reactive-form';
import formInputConfig from './form.config';

const ExampleFeature = () => {
    let genForm;

    const setInputNesting = useCallback(() => {
        genForm
            .get('first_name')
            .onBlurChanges.subscribe((value) => (value ? genForm.patchValue({ full_name: `${value} - ` }) : ''));
        // genForm.valueChanges.subscribe((value) => {
        //     console.log(genForm.status);
        //     if (genForm.status === 'VALID') {
        //         console.log('value', value);
        //         genForm.status = 'INVALID';
        //     }
        // });
        genForm.disable();
    }, [genForm]);

    const handleSubmit = (e) => {
        e.preventDefault();
        alert(`You submitted \n ${JSON.stringify(genForm.value, null, 2)}`);
        console.log(genForm);
    };

    const handleEditMode = () => {
        genForm.enable({ onlySelf: true });
    };

    const handleOnCancelMode = () => {
        genForm.disable({ onlySelf: true });
        genForm.reset();
    };

    const setForm = (form) => {
        genForm = form;
        genForm.meta = {
            handleSubmit: handleSubmit,
            handleEditMode: handleEditMode,
            handleCancelMode: handleOnCancelMode,
        };
    };

    const unSubscribe = useCallback(() => {
        genForm.valueChanges.unsubscribe();
    }, [genForm]);

    useEffect(() => {
        setInputNesting();
        unSubscribe();
    }, [unSubscribe]);

    return (
        <div className="container h-100 w-100 mt-5">
            <div className="row">
                <FormGenerator className="col-12" onMount={setForm} fieldConfig={formInputConfig} />
            </div>
        </div>
    );
};

export default ExampleFeature;
