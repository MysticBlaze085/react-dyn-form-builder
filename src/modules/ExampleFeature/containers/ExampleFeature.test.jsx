import React from 'react';
import { shallow } from 'enzyme';
import ExampleFeature from './ExampleFeature';
import formInputConfig from './form.config';

it('expect CheckBox to render', () => {
    expect(shallow(<ExampleFeature {...formInputConfig} />)).toMatchSnapshot();
});
