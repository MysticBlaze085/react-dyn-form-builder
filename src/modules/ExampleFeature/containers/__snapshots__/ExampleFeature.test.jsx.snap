// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect CheckBox to render 1`] = `
<div
  className="container h-100 w-100 mt-5"
>
  <div
    className="row"
  >
    <FormGenerator
      className="col-12"
      fieldConfig={
        Object {
          "controls": Object {
            "$field_0": Object {
              "isStatic": false,
              "render": [Function],
            },
            "$field_1": Object {
              "isStatic": false,
              "render": [Function],
            },
            "first_name": Object {
              "meta": Object {
                "classes": "col-6 pr-2",
                "label": "First Name",
                "placeholder": "Enter first name",
              },
              "options": Object {
                "validators": [Function],
              },
              "render": [Function],
            },
            "full_name": Object {
              "meta": Object {
                "classes": "col-12 pb-5",
                "label": "Full Name",
                "placeholder": "Enter Full Name",
              },
              "options": Object {
                "validators": [Function],
              },
              "render": [Function],
            },
            "gender": Object {
              "formState": "other",
              "meta": Object {
                "classes": "col-6",
                "items": Array [
                  Object {
                    "key": "male",
                    "value": "Male",
                  },
                  Object {
                    "key": "female",
                    "value": "Female",
                  },
                  Object {
                    "key": "other",
                    "value": "other",
                  },
                ],
                "label": "Gender",
              },
              "options": Object {
                "validators": [Function],
              },
              "render": [Function],
            },
            "last_name": Object {
              "meta": Object {
                "classes": "col-6",
                "label": "Last Name",
                "placeholder": "Enter last name",
              },
              "options": Object {
                "validators": [Function],
              },
              "render": [Function],
            },
            "nationality": Object {
              "meta": Object {
                "classes": "col-6 pr-2",
                "items": Array [
                  Object {
                    "key": "us",
                    "value": "US",
                  },
                  Object {
                    "key": "uk",
                    "value": "UK",
                  },
                  Object {
                    "key": "india",
                    "value": "India",
                  },
                  Object {
                    "key": "china",
                    "value": "China",
                  },
                ],
                "label": "State",
              },
              "render": [Function],
            },
            "notes": Object {
              "meta": Object {
                "label": "Notes",
              },
              "render": [Function],
            },
            "terms": Object {
              "formState": false,
              "meta": Object {
                "classes": "col-6",
                "items": Array [
                  Object {
                    "key": "terms",
                    "value": "I agree to the terms and condition.",
                  },
                ],
              },
              "options": Object {
                "validators": [Function],
              },
              "render": [Function],
            },
          },
        }
      }
      onMount={[Function]}
      onUnmount={[Function]}
    />
  </div>
</div>
`;
